CONTRIBUTING.md

```diff
@@ -457,13 +457,13 @@ func DoFoo() {}
 
 When deprecating a feature affecting end-users, consider first deprecating the feature in one version, then hiding it
 behind a [feature
-flag](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/service/featuregate/)
+gate](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/service/featuregate/)
 in a subsequent version, and eventually removing it after yet another version. This is how it would look like, considering
 that each of the following steps is done in a separate version:
 
-1. Mark the feature as deprecated, add a short lived feature flag with the feature enabled by default
-1. Change the feature flag to disable the feature by default, deprecating the flag at the same time
-1. Remove the feature and the flag
+1. Mark the feature as deprecated, add a short lived feature gate with the feature enabled by default
+1. Change the feature gate to disable the feature by default, deprecating the gate at the same time
```

Comment:
"Deprecating" -> this is marking as stable correct? Also should this be a separate step since we do not allow changing the gate at that point?